httpcore/client.py

```diff
@@ -44,6 +46,8 @@ def __init__(
         )
         self.adapter = EnvironmentAdapter(dispatch=redirect_adapter)
 
+        self.base_url = base_url
```

Comment:
Let's use the `URL` type. Eg. `self.base_url = None if url is None else URL(base_url)`